src/status_im/ui2/screens/common/contact_list/view.cljs

```diff
@@ -0,0 +1,50 @@
+(ns status-im.ui2.screens.common.contact-list.view
+  (:require [react-native.core :as rn]
+            [quo2.foundations.colors :as colors]
+            [status-im.ui2.screens.chat.components.contact-item.view :as contact-item]
+            [quo2.core :as quo2]
+            [utils.re-frame :as rf]
+            [reagent.core :as reagent]
+            [clojure.string :as str]))
+
+(def query (reagent/atom ""))
+
+(defn prepare-contacts [contacts]
+  (let [data (atom {})]
+    (doseq [i (range (count contacts))]
```

Comment:
If we passed in contacts directly, Can't `doseq` iterate over it's count ?

Reply:
This function is too non-idiomatic for Clojure. In Clojure we resist the temptation of processing data in a mutable fashion. At the very least, you can use reduce with the accumulator with a map, but there could be even better ways. It would also perform better because an atom is not like an ordinary variable in Javascript, see the docstring below to get an idea:

> Atomically swaps the value of atom to be: (apply f current-value-of-atom args). Note that f may be called
multiple times, and thus should be free of side effects.  Returns the value that was swapped in.

Reply:
it was fixed here https://github.com/status-im/status-mobile/pull/14419

Reply:
oh ok i see, its list not item, but still i moved this already

Reply:
@ibrkhalil I am not sure what do you mean? It is iterating over contacts count.

Reply:
@ilmotta so in Clojure it is a bad practice to update atoms in a loop?

Reply:
The correct answer is it depends, as usual, but there are almost always better ways. We have to remember that atoms were conceived primarily to manage shared state, and as such, they are capable of a lot of cool stuff, but this also means they're a big bazooka for simple data processing.

Temporary state for data processing in pure functions can almost always be replaced by an *accumulator* in a reducing function (just to name one among several approaches). Using doseq to loop over indexes is quite imperative, and Clojure gives us tools to write more *functional* code.

So something along these lines:

```clojure
;; very imperative
(let [state (atom {})]
  (doseq [i (range (count contacts))
          :let [contact (nth contacts i)]]
    (swap! state ...)))

;; probably better
(reduce (fn [result contact]
          (assoc result ...))
        {}
        contacts)
```